src/rules/cyclomaticComplexityRule.ts

```diff
@@ -80,139 +78,53 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 }
 
-class CyclomaticComplexityWalker extends Lint.RuleWalker {
-
-    private functions: number[] = [];
-
-    public constructor(sourceFile: ts.SourceFile, options: Lint.IOptions, private threshold: number) {
-        super(sourceFile, options);
-    }
-
-    protected visitArrowFunction(node: ts.ArrowFunction) {
-        this.startFunction();
-        super.visitArrowFunction(node);
-        this.endFunction(node);
-    }
-
-    protected visitBinaryExpression(node: ts.BinaryExpression) {
-        switch (node.operatorToken.kind) {
-            case ts.SyntaxKind.BarBarToken:
-            case ts.SyntaxKind.AmpersandAmpersandToken:
-                this.incrementComplexity();
-                break;
-            default:
-                break;
+function walk(ctx: Lint.WalkContext<{ threshold: number }>): void {
+    const { options: { threshold } } = ctx;
+    let complexity = 0;
+
+    return ts.forEachChild(ctx.sourceFile, function cb(node: ts.Node): void {
```

Comment:
That's the way it worked before. Could be updated now, but it would be a breaking change.